### DISCUSSION TITLE
GenMon on New Generac Generators

### DISCUSSION THREAD

**Discussion by redchairlabs** [jgyates/genmon#1177] [General]
I bought my Generac Guardian 22kw generator about six years ago. GenMon has been working reliably ever since I installed it. Fast Forward to now and I my inlawas want to get a standby generator for their house. I want to put a GenMon on it as well but am not sure what the current compatibility is with the newer Generac controllers.

I was hoping someone with recent experience could bring me up to speed.

**Comment by jgyates** [comment]
Genmon works with the latest firmware versions of the Generac Evolution 2.0 controller, which is used on their residential generators.

**Reply by redchairlabs** [reply]
Awesome! I had heard something to the effect that Generac was using a new controller design that didnt have a MODBUS port.

**Reply by biggen1684** [reply]
I bought a new 24kw Generac 3 months ago. Genmon runs flawlessly.

**Reply by zachliu** [reply]
Quick question:
Is it possible for Genmon to increase the Generac (2024 model) exercise time from 5 min to, say 10 min?

**Reply by jgyates** [reply]
For liquid cooled models yes. For air cooled models no. You can use the enhanced exercise add on to create a 2nd exercise cycle controlled by genmon. The controller built in exercise can not be disabled.

**Reply by zachliu** [reply]
Can the genmon-controlled 2nd exercise be more than 5 min?
My neighbor's Kohler has a 20-min weekly exercise, right outside my window. Incredibly annoying.
I really want to fight back by increasing my Generac exercise to 20-min too :smiling_imp:

**Reply by jgyates** [reply]
you can make it any number of minutes. More info is here: https://github.com/jgyates/genmon/wiki/1----Software-Overview#genexercisepy-optional

**Reply by biggen1684** [reply]
@jgyates Will know more but I think if you create your extended exercise time schedule over the built-in exercise time schedule, then you essentially get one exercise time for that period.  Not two distinct times.

**Reply by zachliu** [reply]
@biggen1684 :+1: :+1: :+1: Thanks! Good to know! Getting the HAT now :rocket:

**Reply by jgyates** [reply]
Just to clarify the comment from @biggen1684, the extended exercise has to start before the built in exercise cycle an continue for longer than the built in cycle.

**Reply by redchairlabs** [reply]
> I bought a new 24kw Generac 3 months ago. Genmon runs flawlessly.

Thanks for the feedback!
